flambe/compile/component.py

```diff
@@ -976,8 +976,16 @@ def _state_dict_hook(self,
                                                 prefix=current_path + STATE_DICT_DELIMETER,
                                                 keep_vars=state_dict._metadata[KEEP_VARS_KEY])
                 state_dict._metadata[FLAMBE_DIRECTORIES_KEY].add(current_path)
-        # Iterate over modules to make sure Component
-        # nn.Modules are added to flambe directories
+            # Iterate over modules to make sure NON-Component
+            # nn.Modules' state is added. Only needed if self is not
+            # nn.Module, because otherwise this hook is being called
+            # via nn.Module.state_dict, and will already recurse to
+            # all children modules
+            if ((not isinstance(self, torch.nn.Module)) and isinstance(attr, torch.nn.Module)
```

Comment:
What happens if `self` is a `torch.nn.Module`?

Reply:
Then torch handles the recursion